Artifacts/windows-ServiceFabricSDK/InstallServiceFabricSDKAndTools.ps1

```diff
@@ -1,136 +1,375 @@
+}
+
+function Enable-ServiceFabricTools
+{
+    [CmdletBinding()]
+    param(
+    )
+
     $vsBootstrapperExe = Join-Path $env:Temp "vsbootstrap.exe"
     $vsBootstrapperUrl = 'https://aka.ms/vs/15/release/vs_enterprise.exe'
-    Write-Verbose "Downloading Visual Studio bootstrapper from $vsBootstrapperUrl."
+
+    Write-Host "Downloading Visual Studio bootstrapper from $vsBootstrapperUrl"
     try 
     {
-        (New-Object System.Net.WebClient).DownloadFile($vsBootstrapperUrl, $vsBootstrapperExe )
-    } catch [System.Management.Automation.MethodInvocationException] {
-        Write-Error "Unable to find Visual Studio bootstapper at $vsBootstrapperUrl."
+        (New-Object System.Net.WebClient).DownloadFile($vsBootstrapperUrl, $vsBootstrapperExe)
+    }
+    catch [System.Management.Automation.MethodInvocationException]
+    {
+        throw "Unable to find Visual Studio bootstapper at $vsBootstrapperUrl"
     }
-    if ((Test-Path $vsBootstrapperExe) -eq $false){
-        Write-Error "Visual Studio bootstrapper ($vsBootstrapperUrl) not successfully download to $vsBootstrapperExe."
-        return -1
+
+    if (-not (Test-Path $vsBootstrapperExe))
+    {
+        throw "Visual Studio bootstrapper was not successfully downloaded to $vsBootstrapperExe"
     }
-    StartAndWaitForProcess $vsBootstrapperExe "--quiet --update" @(0)
 
-    #Get path to vs_installer.exe
-    $vsInstallerExePath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{6F320B93-EE3C-4826-85E0-ADF79F8D4C61}' -Name 'InstallLocation').InstallLocation.Trim('"')
+    Write-Host 'Getting Visual Studio Installer path'
+    $vsInstallerExePath = (Get-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{6F320B93-EE3C-4826-85E0-ADF79F8D4C61}" -Name 'InstallLocation').InstallLocation.Trim('"')
     $vsInstallExe = Join-Path $vsInstallerExePath  'vs_installer.exe'
-    Write-Verbose "Visual Studio Installer is at $vsInstallExe"
-  
-   #Find VS Install Path(s) and add Service Fabric component.
-   foreach ($VSInstallPathInfo in $vsInstallInfoArray){
-        Write-Output "Enabling component for Service Fabric Tools for installation $($VSInstallPathInfo.InstallationPath)"
-        StartAndWaitForProcess $vsInstallExe "modify --installPath `"$($VSInstallPathInfo.InstallationPath)`" --add Microsoft.VisualStudio.Component.Azure.ServiceFabric.Tools --add Microsoft.VisualStudio.Workload.Azure --quiet --norestart" @(0)
-    }
-}
\ No newline at end of file
+    Write-Host "Visual Studio Installer is at $vsInstallExe"
+
+    # Get VS installation information.
+    $vsInstances = Get-VSSetupInstances
+
+    # Add Service Fabric component to all Visual Studio instances.
+    $vsInstances | % {
+        if ($_.InstallationVersion.Major -eq 15)
```

Comment:
Good point, I will pass the version number as a parameter, as we should always refrain from using "global" variables, where possible.